mllib/src/main/scala/org/apache/spark/ml/regression/GeneralizedLinearRegression.scala

```diff
@@ -467,10 +469,12 @@ object GeneralizedLinearRegression extends DefaultParamsReadable[GeneralizedLine
 
     override def variance(mu: Double): Double = mu * (1.0 - mu)
 
+    private def ylogy(y: Double, mu: Double): Double = {
+      if (y == 0) 0.0 else y * math.log(y / mu)
+    }
+
     override def deviance(y: Double, mu: Double, weight: Double): Double = {
```

Comment:
Yes, but isn't handling y=0 better than introducing a tiny error in computation? there's no need to avoid computing y ln y at y=0 because it's value is well defined for this purpose. It seems quite less clear to me; I've only ever seen computing y ln y by special-casing y=0 and not restricting its domain.